From: uazo <uazo@users.noreply.github.com>
Date: Thu, 16 Jan 2025 09:43:07 +0000
Subject: Enable Cert Management UI

Activates the ui of certificate management in desktop platforms,
disabling the use of system certificates by default (which can be activated from the ui).

License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
---
 chrome/browser/net/profile_network_context_service.cc      | 7 ++++++-
 .../chrome_features_cc/Enable-Cert-Management-UI.inc       | 5 +++++
 2 files changed, 11 insertions(+), 1 deletion(-)
 create mode 100644 cromite_flags/chrome/common/chrome_features_cc/Enable-Cert-Management-UI.inc

diff --git a/chrome/browser/net/profile_network_context_service.cc b/chrome/browser/net/profile_network_context_service.cc
--- a/chrome/browser/net/profile_network_context_service.cc
+++ b/chrome/browser/net/profile_network_context_service.cc
@@ -541,7 +541,12 @@ void ProfileNetworkContextService::RegisterProfilePrefs(
   registry->RegisterListPref(prefs::kCAHintCertificates);
 #if !BUILDFLAG(IS_CHROMEOS)
   // Include user added platform certs by default.
-  registry->RegisterBooleanPref(prefs::kCAPlatformIntegrationEnabled, true);
+  registry->RegisterBooleanPref(prefs::kCAPlatformIntegrationEnabled,
+#if BUILDFLAG(IS_ANDROID)
+    true);
+#else
+    false);
+#endif
 #endif
 #if BUILDFLAG(IS_CHROMEOS)
   net::ServerCertificateDatabaseService::RegisterProfilePrefs(registry);
diff --git a/cromite_flags/chrome/common/chrome_features_cc/Enable-Cert-Management-UI.inc b/cromite_flags/chrome/common/chrome_features_cc/Enable-Cert-Management-UI.inc
new file mode 100644
--- /dev/null
+++ b/cromite_flags/chrome/common/chrome_features_cc/Enable-Cert-Management-UI.inc
@@ -0,0 +1,5 @@
+#if BUILDFLAG(IS_WIN)
+SET_CROMITE_FEATURE_ENABLED(kEnableCertManagementUIV2);
+SET_CROMITE_FEATURE_ENABLED(kEnableCertManagementUIV2Write);
+SET_CROMITE_FEATURE_ENABLED(kEnableCertManagementUIV2EditCerts);
+#endif
--
